cabal-install/src/Distribution/Client/HttpUtils.hs

```diff
@@ -1073,7 +1080,9 @@ plainHttpTransport =
           setOutHandler (debug verbosity)
           setUserAgent userAgent
           setAllowBasicAuth False
-          setAuthorityGen (\_ _ -> return auth)
+          case auth of
+            Just (Left x) -> setAuthorityGen (\_ _ -> return $ Just x)
+            _ -> setAuthorityGen (\_ _ -> return Nothing)
```

Comment:
Again this picks up `Just (Right...)` and `Nothing`. Are we sure this is correct (and will be correct for the foreseeable future?)

Reply:
There was an issue with the plain-http transport that I have now fixed (along with the wget and powershell transports) but this code is left unchanged. Plain-http has to have username and password set using a function passed to `setAuthorityGen` in the browser. The token has to be set as a header in the request. If the token is set then there won't be a security challenge meaning the function passed to `setAuthorityGen` will never be called by the browser.